### PR TITLE
Refractor json response

### PR DESCRIPTION
**Pull request by ajibukunoluwa**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -12,7 +12,10 @@
     "autoload": {
         "psr-4": {
             "Ichtrojan\\Otp\\": "src/"
-        }
+        },
+        "files": [
+            "src/Helpers/Response.php"
```
Comment (ichtrojan) [inline composer.json]: No helpers, all methods in this file should be in a Class

File: src/Helpers/Response.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            array_merge([
+                'status'    => $status,
+                'message'   => $message
+            ], $additional_data), $code);
+    }
+}
+
+
+
+
```
Comment (ichtrojan) [inline src/Helpers/Response.php]: all these should be in a class

```php
<?php
namespace ...

use ...

class Response {
    public function success {
        ...
    }
}
```

Comment (ichtrojan) [reply src/Helpers/Response.php]: Then you can call it in the `src/Otp.php` file

```php
class Otp {
    public __construct (Response $response) {
        $this->response = $response
    }
}
```

## Usage

```php
$this->response->success($message);
```

File: src/Otp.php [hunk 3]
```diff
@@ -21,73 +21,62 @@ protected static function getFacadeAccessor(): string
...
     {
-        Model::where('identifier', $identifier)->where('valid', true)->delete();
+        Model::identifier($identifier)->valid()->delete();
         $token = str_pad($this->generatePin(), 4, '0', STR_PAD_LEFT);
-        if ($digits === 5) {
+        if (5 === $digits) {
             $token = str_pad($this->generatePin(5), 5, '0', STR_PAD_LEFT);
         }
-        if ($digits === 6) {
+        if (6 === $digits) {
```
Comment (ichtrojan) [inline src/Otp.php]: why?

File: src/Otp.php [hunk 4]
```diff
@@ -21,73 +21,62 @@ protected static function getFacadeAccessor(): string
...
-            return response()->json([
-                'status' => false,
-                'message' => 'OTP does not exist'
-            ]);
+        if (null === $otp) {
+            return abortJson(404, 'OTP does not exist');
         }
 
-        if ($otp->valid === true) {
+        if (true == $otp->valid) {
```
Comment (ichtrojan) [inline src/Otp.php]: why?

Comment (OlabodeAbesin) [reply src/Otp.php]: Hey @ichtrojan 
$otp->valid is stored as 1 in the database.
Unless you cast its data type, doing strict equality will return false.
This should work:

`if ($otp->valid == true) {`

**Pull request closed without merging**